src/main/java/org/sonarlint/intellij/issue/persistence/StoreIndex.java

```diff
@@ -0,0 +1,12 @@
+package org.sonarlint.intellij.issue.persistence;
+
+import java.nio.file.Path;
+import java.util.Collection;
+
+public interface StoreIndex<T> {
+  Collection<T> allStorageKeys();
+
+  void save(Path absoluteMappedPath, T key);
```

Comment:
Instead of `absoluteMappedPath`, how about simply `path` ? Also, I think it would be more natural to flip the order of arguments